aspnet/migration/http-modules/sample/Asp.Net5/src/Asp.Net5/Middleware/AddLinesMiddleware.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Http;
+using System.Threading.Tasks;
+
+using System.Text;
+
+namespace Asp.Net5.Middleware
+{
+    public class AddLinesMiddleware
+    {
+        private readonly RequestDelegate _next;
+
+        public AddLinesMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext context)
```

Comment:
No no no no no, this may work in some scenarios, but it will break others horribly. E.g. this will cause the static file middleware to throw, and probably MVC as well.

General middleware guildlines:
Produce a response OR call next.
Do not start a response and then call next.
Be very careful about touching the response after calling next. E.g. you may no longer be able to add headers if they have already been sent (Response.HasStarted), since there is no default response buffering anymore.
You should avoid adding to someone else's response body haphazardly. E.g. if they set the content-length header then you'll be writing off the end of the response.
If you want to post-process a response then you should buffer it yourself (e.g. replace the response stream with a buffer), call next, do your post-processing, and then send it.